### PR TITLE
Query for all opportunities

### PR DESCRIPTION
**Pull request by Adzz**

### CODE REVIEW

File: lib/closex/http_client.ex [hunk 1]
```diff
@@ -95,12 +76,47 @@ defmodule Closex.HTTPClient do
...
+  def find_all_opportunities(term, limit \\ 100) do
+    find_all("opportunity", term, limit)
   end
 
   # Private stuff...
 
+  defp find(resource, search_term, opts \\[]) do
+    opts = merge_search_term_into_opts(search_term, opts)
+
+    # HACK: Pass empty object as body to workaround bug in HTTPoison, see: https://github.com/edgurgel/httpoison/issues/285
```
Comment (sadir) [inline lib/closex/http_client.ex]: The issue looks to be fixed now! Can we delete this comment or do we need to change some code?

File: test/http_client_test.exs [hunk 2]
```diff
@@ -490,20 +502,22 @@ defmodule Closex.HTTPClientTest do
     end
   end
   describe "get_users/0" do
+    @tag :this
```
Comment (sadir) [inline test/http_client_test.exs]: 🔫

File: test/http_client_test.exs [hunk 3]
```diff
@@ -522,4 +536,16 @@ defmodule Closex.HTTPClientTest do
       end
     end
   end
+
+  describe "get_all/2" do
+    test "find_all_opportunities/2" do
+      use_cassette "find_all_opportunities", match_requests_on: [:query] do
+        number_of_opportunities_in_closeio_dev_account = 65
```
Comment (sadir) [inline test/http_client_test.exs]: 👍

File: test/http_client_test.exs (hunk 2 above)
Comment (samphilipd) [reply test/http_client_test.exs]: ^^ I hate how on macs this renders as a water pistol :/

**Pull request merged**